### ISSUE TITLE
Compiler warnings for headers in subprojects

### ISSUE THREAD

**Issue by NickeZ** [mesonbuild/meson#3197]
I noticed that I get compiler warnings for headers in subprojects. Is there a way to mark include directories from subprojects as `is_system`?

If this isn't supported, how would you implement it and would you accept a pull request?

**Comment by jpakkane** [comment]
We have toyed with it but the problem is that using -isystem instead of -I yields a different search path which can break things.

You can specify `is_system` manually on specific dirs if you choose to, though. See the docs.

**Comment by nirbheek** [comment]
Yes, you usually want to search subproject include paths before the system ones, and that would break that.

**Comment by NickeZ** [comment]
> You can specify is_system manually on specific dirs if you choose to, though. See the docs.

Yeah, I understand that I can go in an modify all my dependencies. Is there a way to do that from the main `meson.build` file?

**Comment by jpakkane** [comment]
No.

**Comment by nirbheek** [comment]
We could probably add a kwarg to `declare_dependency()` to do that, which would also apply transitively to dependencies inside it. But then we get to very edge-case territory. I'm sure that when some other tangentially-related use-case crops up we'll see a better way to solve this.

**Comment by NickeZ** [comment]
Opening again because this isn't true:

> We have toyed with it but the problem is that using -isystem instead of -I yields a different search path which can break things.

There are no practical differences between `-isystem` and `-I` when it comes to searching for headers compared to the system headers. Subprojects headers would have lower precedence than current project headers. But I think that is fair?

According to docs:

No difference in how you can use them:

> Directories specified with -I, -isystem, or -idirafter apply to lookup for both the #include "file" and #include \<file\> directives. 

`-isystem` is lower than `-I` but still higher than system paths:

> 3. Directories specified with -I options are scanned in left-to-right order.
> 4. Directories specified with -isystem options are scanned in left-to-right order.
> 5. Standard system directories are scanned. 

This isn't supported on MSVC yet. (I think experimental support exists)
